nemo/core/connectors/save_restore_connector.py

```diff
@@ -122,9 +122,11 @@ def restore_from(
                         # Resolve the override config
                         conf = OmegaConf.to_container(conf, resolve=True)
                         conf = OmegaConf.create(conf)
+
                 # If override is top level config, extract just `model` from it
-                if 'model' in conf:
```

Comment:
@ericharper 
Hi Eric,
Can you please comment on necessity of this two lines.

The problem for me right now is following:
When we create ModelPT class, we pass only model part of it. So, when doing automatic saving while checkpointing, only this part is saved within .nemo file.
So, if your ModelPT has inner object with the same name 'model' (as well as config part), current code doens't distinct it from outer config and make `conf = conf.model` assignment.
If I comment this two lines, some tests become invalid, i.e. tests/core/test_save_restore.py::TestSaveRestore::test_mock_restore_from_config_override_with_yaml

Reply:
Basically, current NeMo inner implementation forbids to create nn.Modules with plain 'model' naming, which is quite common to use in practice.

Reply:
We use `self.model` here: https://github.com/NVIDIA/NeMo/blob/d5bd93675b82045d1ca7541e45981fe0440631a6/nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py#L86

Reply:
@titu1994

Reply:
So this problem is coming from the config having `model.model` ? https://github.com/NVIDIA/NeMo/blob/d5bd93675b82045d1ca7541e45981fe0440631a6/examples/tts/conf/talknet-durs.yaml#L66

Instead of commenting those lines, can you just check for a top level key or something like that?

Reply:
I really can't come up with definite way to separate config from being top-level vs inner node.
1) There is no parent link in OmegaConf.
2) A check like `'name' in conf` or `'sample_rate' in conf` should work for almost all cases, but there is always test case against it.

I suggest maybe we should't able to load models and overriding restoration with top-level configs. Model is something that have access to only 'model' part of initial config, so when loading it should only operate with `conf.model` part of it.

Reply:
Woops sorry misclick. 
Yes only cfg.model will be saved in Nemo config. Yes, you cannot have cfg.model.model in your config name. It is because save restore does not know if config is from user in constrictor of from file inside .Nemo so it needs to bring "cfg.model" to just model level so that OmegaConf resolution works properly. 

It won't be reverted so I suggest changing cfg.model.model to cfg.model.<some other name>. It is anyway awkward having cfg.model.model. The outer model represents the whole model, therefore inner model name makes no sense. You can call if module if you absolutely want to, or pick another name.

Reply:
> I suggest maybe we should't able to load models and overriding restoration with top-level configs.

This is required in useful cases, so it will be maintained.